lib/rucio/api/authorisation.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2012-2018 CERN for the benefit of the ATLAS collaboration.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Authors:
+# - Mario Lassnig <mario@lassnig.net>, 2018
+
+from rucio.api import permission
+from rucio.common import exception
+from rucio.core import authorisation
+
+
+def get_signed_url(account, appid, ip=None, service=None, operation=None, url=None):
+    """
+    Get a signed URL for a particular service and operation.
+
+    The signed URL will be valid for 1 hour.
+
+    :param account: Account identifier as a string.
+    :param appid: The application identifier as a string.
+    :param ip: IP address of the client as a string.
+    :param service: The service to authorise, currently only 'gsc'.
```

Comment:
Remark: would be good  to have a look on s3 and swift for a common signed url terminology.

Reply:
From what I've seen SciTokens, GCS, and S3 each have their own input and output parameters... At least S3 and Swift are the same ;-)